components/google-cloud/google_cloud_components/aiplatform/utils.py

```diff
@@ -280,6 +281,90 @@ def key(param):
     )
 
 
+def filter_docstring_args(
+    signature: inspect.Signature,
+    docstring: str,
+    is_init_signature: bool = False,
+) -> Dict[str, str]:
+    """Removes unused params from docstring Args section.
+
+    Args:
+        signature (inspect.Signature): Model Builder SDK Method Signature.
+        docstring (str): Model Builder SDK Method docstring from method.__doc__
+        is_init_signature (bool): is this constructor signature
+
+    Returns:
+        Dictionary of Arg names as keys and descriptions as values.
+    """
+    try:
+        parsed_docstring = docstring_parser.parse(docstring)
+    except ValueError:
+        return {}
+    args_dict = {p.arg_name: p.description for p in parsed_docstring.params}
+
+    new_args_dict = {}
+    for param in signature.parameters.values():
+        if param.name not in PARAMS_TO_REMOVE:
+            new_arg_name = param.name
+            # change resource name signatures to resource types
+            # to match new param.names ie: model_name -> model
+            if is_init_signature and is_resource_name_parameter_name(param.name
+                                                                    ):
+                new_arg_name = param.name[:-len('_name')]
+
+            # check if there was an arg description for this parameter.
+            if args_dict.get(param.name, False):
+                new_args_dict[new_arg_name] = args_dict.get(param.name)
+    return new_args_dict
+
+
+def generate_docstring(
+    args_dict: Dict[str, str], signature: inspect.Signature,
+    method_docstring: str
+) -> str:
+    """Generates a new doc string using args_dict provided.
+
+    Args:
+        args_dict (Dict[str, str]): A dictionary of Arg names as keys and descriptions as values.
+        signature (inspect.Signature): Method Signature of the converted method.
+        method_docstring (str): Model Builder SDK Method docstring from method.__doc__
+    Returns:
+        A doc string for converted method.
+    """
+    try:
+        parsed_docstring = docstring_parser.parse(method_docstring)
+    except ValueError:
```

Comment:
It would be good to know which docstrings are failing as it may be on the MB SDK team to format the docstring.

Reply:
these are the ones that fail: 
```
AutoMLTabularTrainingJobRunOp(...)
AutoMLTextTrainingJobRunOp(...)
CustomContainerTrainingJobRunOp(...)
CustomPythonPackageTrainingJobRunOp(...)
ModelUploadOp(...)
```